crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -424,12 +439,20 @@ async fn execute_transactions(
                 periods += 1;
             }
             Ok(Err(err)) => return Err(err),
-            Ok(Ok(_)) => {
+            Ok(Ok(effects)) => {
                 authority_store.insert_executed_transactions(
                     &all_tx_digests,
                     epoch_store.epoch(),
                     checkpoint_sequence,
                 )?;
+
+                let state = CheckpointExecutionState {
```

Comment:
approach here is the same as in https://github.com/MystenLabs/sui/pull/7399. 
eventually we should unify them